src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj

```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);net461;netstandard2.0;netstandard2.1</TargetFrameworks>
+    <ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage>
```

Comment:
I'm noticing some code under `#if netcoreapp` in this library that will never ship now.  
https://github.com/dotnet/runtime/blob/c64ebe622d187463c8aa9a38fe31ea1dc5485383/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteRuntimeResolver.cs#L45-L47
https://github.com/dotnet/runtime/blob/c64ebe622d187463c8aa9a38fe31ea1dc5485383/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/Expressions/ExpressionResolverBuilder.cs#L74-L78

Reply:
This one is super important, as it allows for DI to be trimmed in Blazor WASM:

https://github.com/dotnet/runtime/blob/0026a202ab4087d237f6be8824dff322d3a7d6be/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceCollectionContainerBuilderExtensions.cs#L63-L75

Reply:
Sounds like we want to keep the net5.0 configuration for DI for the optimized code path?

Reply:
the DI assembly being used in Blazor WASM is the $(NetCoreAppCurrent) one that we hand over to aspnetcore via the transport package or the net5.0 configuration in the DI package?

Reply:
Either that or fix all the places that `#if NETCOREAPP` in this code and change to `#if NETSTANDARD2_1`. And if the APIs aren't there, then yes, we will need to keep an NETCOREAPP build for this library.

Reply:
Today when I build/trim a Blazor WASM app, this is the file that is used:

```
-reference "...\.nuget\packages\microsoft.extensions.dependencyinjection\6.0.0-alpha.1.21070.2\lib\net6.0\Microsoft.Extensions.DependencyInjection.dll"
```

Reply:
That is the `$(NetCoreAppCurrent)` configuration of the package then which definitely shouldn't be part of the package 🗡️. I will see if netstandard2.1 suffices, otherwise I will keep the net5.0 configuration.

Reply:
Oh nice, netstandard2.1 worked.

Reply:
> is based on the $(NetCoreAppCurrent) configuration that we hand over to aspnetcore via the transport package 

For my understanding - why are we still building this assembly for `NetCoreAppCurrent`? Why not just build for `netstandard2.1` and hand that over to aspnetcore via the transport package?

Reply:
I have absolutely no idea... @maryamariyan @ericstj?

Reply:
> For my understanding - why are we still building this assembly for NetCoreAppCurrent? Why not just build for netstandard2.1 and hand that over to aspnetcore via the transport package?

I'm not sure either. 
Perhaps it should be fine to remove NetCoreAppCurrent now once the places with `#if NETCOREAPP` are fixed, to just use a netstandard2.1 condition.

Just FYI: This was the PR which added NetCoreAppCurrent while enabling DI in runtime originally: https://github.com/dotnet/runtime/pull/33816
With diff in `src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj`:
![image](https://user-images.githubusercontent.com/5897654/108137803-7a46ae80-7071-11eb-8a30-b3f892973cc5.png)

Reply:
So we've had 2 reasons for a non-packaged "netcoreappcurrent" configuration in the past.
1. The library is also in the shared framework and needs to be lower in the stack (eg below netstandard.dll).  Not the case here.
2. Due to the partial-build of verticals and the "ref" pass that used to be required for building any targetframework we needed a configuration that would build without requiring more sequential build steps.
3. Source build requires a "source buildable" configuration.

This was probably added for some version of reason 2. @ViktorHofer I suspect that your work to enable packagereference and project reference for most oob libraries has eliminated the need for this reason.  

Also, we've previously discussed that we could probably remove reason 3 if we just enable allconfigurations for sourcebuild.  I think the last blocker there was the partial-reference-facades for .NETFramework that used ILRewriting.

Given both of those, we could probably remove nearly all cases of ExcludeCurrentXXXPackage (keeping only the ones for case 1).

Reply:
> Source build requires a "source buildable" configuration.

Why doesn't that apply to _all_ assemblies? For example in Extensions: Caching, Configuration, FileProviders, and Hosting don't target any .NET Core TFM - just netstandard and netfx.


> we could probably remove nearly all cases of ExcludeCurrentXXXPackage

🚀

Reply:
> Why doesn't that apply to all assemblies? For example in Extensions: Caching, Configuration, FileProviders, and Hosting don't target any .NET Core TFM - just netstandard and netfx.

.NETStandard and .NETFramework tfms are already source buildable but previous .NETCoreApp versions aren't yet. That means that by removing the `$(NetCoreAppCurrent)` configuration, the `netcoreapp3.x` tfm would be built in source build which would fail.

That said, that shouldn't be a concern, as we don't support source building anything else than `$(NetCoreAppCurrent)` in the repository.

Reply:
Removed the configuration.

Reply:
> Why doesn't that apply to all assemblies? For example in Extensions: Caching, Configuration, FileProviders, and Hosting don't target any .NET Core TFM - just netstandard and netfx.

It applies to all assemblies. I wasn’t listing the specific reasons for this assembly but the reasons why any assembly might need to include non-package configurations.

Reply:
> That said, that shouldn't be a concern, as we don't support source building anything else than $(NetCoreAppCurrent) in the repository.

We produce packages during source build today and those packages contain netcoreapp and netstandard assets.